src/task_types/resmoke_tasks.rs

```diff
@@ -463,8 +435,14 @@ impl GenResmokeTaskServiceImpl {
             let test_name = get_test_name(&test);
             if let Some(test_stats) = task_stats.test_map.get(&test_name) {
                 running_runtimes[min_idx] += test_stats.average_runtime;
+                running_tests[min_idx].push(test.clone());
+            } else {
+                left_tests.push(test.clone());
             }
-            running_tests[min_idx].push(test.clone());
+        }
+
+        for (i, test) in left_tests.iter().enumerate() {
```

Comment:
Should we start at the `running_tests` with the minimal sum of run time when we start processing `left_tests`?

Reply:
Added.